kroxylicious-filters/kroxylicious-record-validation/src/test/java/io/kroxylicious/proxy/filter/schema/JsonSchemaBytebufValidatorTest.java

```diff
@@ -98,63 +98,96 @@ public static void shutdownMockRegistry() {
+                .returns(false, Result::valid);
     }
 
     @Test
-    void testValueInvalidAgeInvalidated() {
-        Record record = createRecord("a", "{\"firstName\":\"a\",\"lastName\":\"b\",\"age\":-3}");
-        BytebufValidator validator = BytebufValidators.jsonSchemaValidator(Map.of("apicurio.registry.url", registryServer.baseUrl()), 1L);
-        Result result = validate(record, validator);
-        assertFalse(result.valid());
-        verifyNoMoreInteractions(mockValidator);
+    void valueWithCorrectSchemaIdInHeaderPassesValidation() {
+        Header[] headers = new Header[]{ new RecordHeader(SerdeHeaders.HEADER_VALUE_GLOBAL_ID, toByteArray(GLOBAL_ID)) };
+        Record record = record(RECORD_KEY, VALID_JSON, headers);
+        BytebufValidator validator = BytebufValidators.jsonSchemaValidator(apicurioConfig, GLOBAL_ID);
+        var future = validator.validate(record.value(), record, false);
+        assertThat(future)
+                .succeedsWithin(Duration.ofSeconds(1))
+                .returns(true, Result::valid);
     }
 
     @Test
-    void testInvalidValueInValidated() {
-        Record record = createRecord("a", "123");
-        BytebufValidator validator = BytebufValidators.jsonSchemaValidator(Map.of("apicurio.registry.url", registryServer.baseUrl()), 1L);
-        Result result = validate(record, validator);
-        assertFalse(result.valid());
-        verifyNoMoreInteractions(mockValidator);
+    void valueWithCorrectSchemaIdInBodyPassesValidation() {
+        var value = asSchemaIdPrefixBuf(GLOBAL_ID, VALID_JSON);
+        Record record = record(RECORD_KEY, value);
+        BytebufValidator validator = BytebufValidators.jsonSchemaValidator(apicurioConfig, GLOBAL_ID);
+        var future = validator.validate(record.value(), record, false);
+        assertThat(future)
+                .succeedsWithin(Duration.ofSeconds(1))
+                .returns(true, Result::valid);
     }
 
-    private static Result validate(Record record, BytebufValidator validator) {
-        try {
-            return validator.validate(record.value(), record, false).toCompletableFuture().get(5, TimeUnit.SECONDS);
-        }
-        catch (ExecutionException | InterruptedException | TimeoutException e) {
-            throw new RuntimeException(e);
-        }
+    @Test
+    void valueWithWrongSchemaIdInHeaderRejected() {
+        Header[] headers = new Header[]{ new RecordHeader(SerdeHeaders.HEADER_VALUE_GLOBAL_ID, toByteArray(GLOBAL_ID + 1)) };
+        Record record = record(RECORD_KEY, VALID_JSON, headers);
+        BytebufValidator validator = BytebufValidators.jsonSchemaValidator(apicurioConfig, GLOBAL_ID);
+        var future = validator.validate(record.value(), record, false);
+        assertThat(future)
+                .succeedsWithin(Duration.ofSeconds(1))
+                .returns(false, Result::valid)
+                .returns("Unexpected schema id in record (2), expecting 1", Result::errorMessage);
```

Comment:
I would probably have written:
```suggestion
        assertThat(future)
                .succeedsWithin(Duration.ofSeconds(1))
                .isEqualTo(new Result(false, "Unexpected schema id in record (2), expecting 1"));
```
which I'm not sure is actually any more readable, but I'm curious if you had a reason for using `returns` or a reason not to use `isEqualTo` (no is perfectly legit and I'm not requesting a change).

Reply:
another option:

```
.satisfies(result -> {
                    assertThat(result.valid()).isFalse();
                    assertThat(result.errorMessage()).isEqualTo("Unexpected schema id in record (2), expecting 1");
                });
```